### PR TITLE
discovery: Use pkg/languagedetection for cmdline-based detection

### PR DESCRIPTION
**Pull request by vitkyrka**
### What does this PR do?

discovery: Use pkg/languagedetection for cmdline-based detection

pkg/languagedetection also has cmdline-based language detection; use
that instead of the custom implementation in service discovery to avoid
code duplication.

The main difference is that pkg/languagedetection only looks at the
binary name while the existing code in service discovery looks at all
arguments. From the tests, it looks like the reason for that was to
classify things like "time /usr/bin/java ..." as Java. However, if such a
command is executed, time(1) (or other such processes, such as sudo(8)
or env(1)) will fork and the actual process running java will have
"/usr/bin/java ..." as the command line.

Also, looking at the entire command line leads to false positives with things
like "aws-ebs-csi-driver node --enable-otel-tracing ..".

### Motivation

https://datadoghq.atlassian.net/browse/USMON-1155

### CODE REVIEW

File: pkg/collector/corechecks/servicediscovery/language/language.go [hunk 1]
```diff
@@ -146,12 +150,21 @@ func FindInArgs(args []string) Language {
...
-		}
+
+	langs := languagedetection.DetectLanguage([]languagemodels.Process{&procutil.Process{
+		// Pid doesn't matter since sysprobeConfig is nil
+		Pid:     0,
+		Cmdline: args,
+		Comm:    args[0],
+	}}, nil)
+
+	lang := langs[0]
```
Comment (guyarb) [inline pkg/collector/corechecks/servicediscovery/language/language.go:161]: is it safe to assume we always have at least one entry?

**Review by guyarb — APPROVED** [review]

File: pkg/collector/corechecks/servicediscovery/language/language.go (hunk 1 above)
Comment (vitkyrka) [reply pkg/collector/corechecks/servicediscovery/language/language.go:164]: Currently `DetectLanguage` always returns the correct number of entries, but I've added a check above for safety.

File: pkg/collector/corechecks/servicediscovery/language/language.go [hunk 2]
```diff
@@ -146,12 +150,24 @@ func FindInArgs(args []string) Language {
...
-		// if procName is a known language, return the pos and the language
-		if lang, ok := procToLanguage[procName]; ok {
-			return lang
-		}
+
+	langs := languagedetection.DetectLanguage([]languagemodels.Process{&procutil.Process{
+		// Pid doesn't matter since sysprobeConfig is nil
+		Pid:     0,
+		Cmdline: args,
+		Comm:    args[0],
```
Comment (vitkyrka) [inline pkg/collector/corechecks/servicediscovery/language/language.go:158]: Added a ticket to investigate changing this to exe as a future improvement: https://datadoghq.atlassian.net/browse/USMON-1206

**Pull request merged**